ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -166,6 +182,10 @@ void shutdown() throws InterruptedException {
       final AtomicInteger numFailures = new AtomicInteger(0);
       List<Thread> shutdownThreads = new ArrayList<>();
       for (final Map.Entry<PartitionId, BlobStore> partitionAndStore : stores.entrySet()) {
+        if (stoppedReplicas.contains(partitionAndStore.getKey().toPathString())) {
```

Comment:
Instead of this check (which may differ if the blob store was started but failed to be removed from the stopped list), isn't it just better to do
```
if (!partitionAndStore.getValue().isStarted()) {
  continue;
}
````
?

Reply:
yes, good point. Will take your advice.